config/routes.rb

```diff
@@ -7,6 +7,7 @@
         get "/find", to: "find_merchants#show"
         get "/find_all", to: "find_merchants#index"
         get "/random", to: "random_merchants#show"
+        get "/:id/revenue", to: "merchants_revenue#show"
       end
```

Comment:
This to me feels like a nested resource, so my recommendation would be to use that pattern.